AraCorrelator/RayTraceCorrelator.cxx

```diff
@@ -162,78 +133,56 @@ void RayTraceCorrelator::LoadArrivalTimeTables(const std::string &filename, int
-    tTree -> SetBranchAddress("thetaBin", & thetaBin);
-    tTree -> SetBranchAddress("arrivalTime", & arrivalTime);
-    tTree -> SetBranchAddress("arrivalTheta", & arrivalTheta);
-    tTree -> SetBranchAddress("arrivalPhi", & arrivalPhi);
-    tTree -> SetBranchAddress("launchTheta", & launchTheta);
-    tTree -> SetBranchAddress("launchPhi", & launchPhi);    
-    tTree -> SetBranchAddress("phi", & phi);
-    tTree -> SetBranchAddress("theta", & theta);
-
-    // and put those values into the vector
-    int nEntries = tTree -> GetEntries();
-    for (int i = 0; i < nEntries; i++) {
-        tTree -> GetEntry(i);
-        arrivalTimes_[solNum][thetaBin][phiBin][ant] = arrivalTime;
-        arrivalThetas_[solNum][thetaBin][phiBin][ant] = arrivalTheta;
-        arrivalPhis_[solNum][thetaBin][phiBin][ant] = arrivalPhi;
-        launchThetas_[solNum][thetaBin][phiBin][ant] = launchTheta;
-        launchPhis_[solNum][thetaBin][phiBin][ant] = launchPhi;        
+    std::map<int, TH2D> this_arrivalTimes;
+    std::map<int, TH2D> this_arrivalThetas;
+    std::map<int, TH2D> this_arrivalPhis;
+    std::map<int, TH2D> this_launchThetas;
+    std::map<int, TH2D> this_launchPhis;
+
+    for(int i=0; i<numAntennas_; i++){
+        std::   stringstream ss;
+
+        ss.str(""); ss << "arrival_time_ch_" << i;
+        TH2D *hArrivalTime = (TH2D*)((TH2D*)infile->Get(ss.str().c_str())->Clone()); // clone, so this survives the file closure
+        this_arrivalTimes[i] = *hArrivalTime; // de ref the pointer; give me the memory
+
+        ss.str(""); ss << "arrival_theta_ch" << i;
+        TH2D *hArrivalTheta = (TH2D*)((TH2D*)infile->Get(ss.str().c_str())->Clone()); // clone, so this survives the file closure
+        this_arrivalThetas[i] = *hArrivalTheta; // de ref the pointer; give me the memory
+
+        ss.str(""); ss << "arrival_phi_ch" << i;
+        TH2D *hArrivalPhi = (TH2D*)((TH2D*)infile->Get(ss.str().c_str())->Clone()); // clone, so this survives the file closure
+        this_arrivalPhis[i] = *hArrivalPhi; // de ref the pointer; give me the memory
+
+        ss.str(""); ss << "launch_theta_ch" << i;
+        TH2D *hLaunchTheta = (TH2D*)((TH2D*)infile->Get(ss.str().c_str())->Clone()); // clone, so this survives the file closure
+        this_launchThetas[i] = *hLaunchTheta;// de ref the pointer; give me the memory
+        
+        ss.str(""); ss << "launch_phi_ch" << i;
+        TH2D *hLaunchPhi = (TH2D*)((TH2D*)infile->Get(ss.str().c_str())->Clone()); // clone, so this survives the file closure
+        this_launchPhis[i] = *hLaunchPhi; // de ref the pointer; give me the memory
     }
+    arrivalTimes_[solNum] = this_arrivalTimes;
+    arrivalThetas_[solNum] = this_arrivalThetas;
+    arrivalPhis_[solNum] = this_arrivalPhis;
+    launchThetas_[solNum] = this_launchThetas;
+    launchPhis_[solNum] = this_launchPhis;
 
     // close up
     infile->Close();
-}
 
-RayTraceCorrelator::~RayTraceCorrelator()
```

Comment:
There's still a destructor right? The diff is just off?

Reply:
Yes, it definitely still has one. Not sure why it's made about this.

Reply:
Okay cool, no worries. Too many big changes to track probably haha